packages/sitecore-jss-nextjs/src/components/NextImage.test.tsx

```diff
@@ -0,0 +1,323 @@
+const customLoader = ({ src }) => new URL(`https://cm.jss.localhost${src}`).href;
+const mockLoader = (spy(customLoader) as unknown) as MockLoaderType;
+
+const loaderParams = {
+  config: {
+    allSizes: [16, 32, 48, 64, 96, 128, 256, 384, 640, 750, 828, 1080, 1200, 1920, 2048, 3840],
+    // eslint-disable-next-line quotes
+    contentSecurityPolicy: "script-src 'none'; frame-src 'none'; sandbox;",
+    dangerouslyAllowSVG: false,
+    deviceSizes: [640, 750, 828, 1080, 1200, 1920, 2048, 3840],
+    disableStaticImages: false,
+    domains: [],
+    formats: ['image/webp'],
+    imageSizes: [16, 32, 48, 64, 96, 128, 256, 384],
+    loader: 'default',
+    minimumCacheTTL: 60,
+    path: '/_next/image',
+  },
+  quality: 75 || undefined,
+  src: '/assets/img/test0.png',
+  width: 8,
+};
+
+const expect = chai.use(chaiString).expect;
+describe('<NextImage />', () => {
+  describe('Next loader function', () => {
+    it('should append configPath when src is relative', () => {
+      const params: ImageLoaderProps = {
+        config: {
+          deviceSizes: [],
+          imageSizes: [],
+          path: 'https://cm.jss.localhost',
+          allSizes: [],
+          loader: 'default',
+          domains: [],
+          disableStaticImages: false,
+          minimumCacheTTL: 1,
+          formats: [],
+          dangerouslyAllowSVG: false,
+          contentSecurityPolicy: 'test',
+        },
+        src: '/assets/img/test0.png',
+        width: 100,
+      };
+      const result = loader(params);
+      expect(result).to.be.a('string');
+      expect(result).to.equal(`${params.config.path}/assets/img/test0.png?mw=100`);
+    });
+
+    it('should not require config path when src is absolute', () => {
+      const params: ImageLoaderProps = {
+        config: {
+          deviceSizes: [],
+          imageSizes: [],
+          path: undefined,
+          allSizes: [],
+          loader: 'default',
+          domains: [],
+          disableStaticImages: false,
+          minimumCacheTTL: 1,
+          formats: [],
+          dangerouslyAllowSVG: false,
+          contentSecurityPolicy: 'test',
+        },
+        src: 'https://cm.jss.localhost/assets/img/test0.png?mw=100',
+        width: 100,
+      };
+      const result = loader(params);
+      expect(result).to.be.a('string');
+      expect(result).to.equal('https://cm.jss.localhost/assets/img/test0.png?mw=100');
+    });
+
+    it('should not append config path when src is absolute', () => {
+      const params: ImageLoaderProps = {
+        config: {
+          deviceSizes: [],
+          imageSizes: [],
+          path: 'https://cm.jss.localhost/',
+          allSizes: [],
+          loader: 'default',
+          domains: [],
+          disableStaticImages: false,
+          minimumCacheTTL: 1,
+          formats: [],
+          dangerouslyAllowSVG: false,
+          contentSecurityPolicy: 'test',
+        },
+        src: 'https://cm.jss.localhost/assets/img/test0.png?mw=100',
```

Comment:
Drop the query string (wouldn't be here)
```suggestion
        src: 'https://cm.jss.localhost/assets/img/test0.png',
```

Reply:
The loader function returns the url with the query string.